### DISCUSSION TITLE
Seqr performance - multiple queries on same project at the same time

### DISCUSSION THREAD

**Discussion by jxchong** [broadinstitute/seqr#2402] [Q&A] [answered]
If we have a WGS dataset (let's say just a trio) and have multiple people performing variant queries/filtration and browsing through the results and are experiencing issues where one user's query will not complete (spinning for >1 minute) while the others are working fine, does that indicate we need to add resources to our instance?

**Comment by hanars** [comment, marked as answer]
So it might mean you need a more robust elasticsearch instance with more client/ master nodes. It might also mean that you just need more seqr processes available, which is set via the `GUNICORN_WORKER_THREADS` environment variable. This determines how many concurrent requests seqr can handle, so if all of them are handling slow searches other users may see the issues you describe. The default for that variable in docker-compose is 4, our production instance uses 20. Upping that number may require more memory to be available for whatever machine is running the seqr component.
